LayoutTests/inspector/css/setLayoutContextTypeChangedMode.html

```diff
@@ -27,6 +30,27 @@
         return Array.from(WI.domManager.attachedNodes({filter: (node) => node.layoutContextType === type}));
     }
 
+    async function domNodeWithIdLayoutFlagsChanged(id)
+    {
+        console.assert(!_test_oneTimeLayoutChangePromisesById.has(id));
+
+        let pendingWrappedPromise = new WI.WrappedPromise;
```

Comment:
NIT: if all you really need to save is the `resolve`, you could just
```
return new Promise((resolve, reject) => {
    _test_oneTimeLayoutChangePromisesById.set(id, resolve);
});
```

also, I think you could restructure this whole thing into one function
```
function awaitLayoutFlagsChanged(selector) {
    return new Promise((resolve, reject) => {
        WI.DOMNode.addEventListener(WI.DOMNode.Event.LayoutFlagsChanged, function listener(event) {
            if (event.target.escapedIdSelector !== selector)
                return;

            WI.DOMNode.removeEventListener(WI.DOMNode.Event.LayoutFlagsChanged, listener);
            resolve();
        });
    });
}
```